File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = 0
    if s[0] == ')':
        return False
    
    for i in range(len(s)):
        if s[i] == '(':
            answer += 1
        elif s[i] == ')':
            answer -= 1 
            if answer < 0:
                return False
    
    return not answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let answer = 0;
    if (s[0] == ')') return false
    
    for(let i = 0; i < s.length; i++){        
        if(s[i] == '('){
            answer += 1
        }
        else if(s[i] == ')'){
            answer -= 1 
            if(answer < 0) return false
        }
    }    
    return !answer 
}